scout/server/blueprints/variants/controllers.py

```diff
@@ -1488,7 +1489,38 @@ def persistent_filter_actions(
         filter_id = request_form.get("filters")
         filter_obj = store.retrieve_filter(filter_id)
         if filter_obj is not None:
-            form = FiltersFormClass(MultiDict(filter_obj))
+
+            def _coerce_formdata(filter: dict) -> MultiDict:
+                """According to wtforms docs https://wtforms.readthedocs.io/en/3.1.x/forms/#the-form-class,
+                formdata passed to Form init needs to be pre-coerced to types. Hence especially the replace
+                of locale specific chars in numbers on class process_formdata would not be enough for the
+                better_decimal form fields.
+                """
+                for filter_decimal_value_form_field in [
+                    "cadd_score",
+                    "revel",
+                    "gnomad_frequency",
+                    "control_frequency",
+                    "tumor_frequency",
+                ]:
+                    if (
```

Comment:
You can skip the try and catch if you check that the string is a number.

Something like this perhaps?

```
if filter_decimal_value_form_field not in filter:
  continue

filter_value = filter_decimal_value_form_field][0]

if isinstance(filter_value, str) and filter_value.replace(",","").isnumeric(): # note that I'm replacing comma with empty string
```